### PR TITLE
Multiple workspaces support.

### PR DESCRIPTION
**Pull request by larskendall**
You should now be able to select multiple workspaces in the Edit Role dialog. When you click Analytics in the sidebar, you'll now be taken to a page that lists all of your workspaces. Clicking any of the cards should take you to that particular workspace in Kibana.

Screen Shot 2021-03-11 at 2 22 50 PM

Screen Shot 2021-03-12 at 8 48 14 AM

Screen Shot 2021-03-12 at 8 48 21 AM

### CODE REVIEW

**Comment by larskendall** [comment]
I put a fix in for that whitespace issue. Ended up avoiding using Grid and just used flexbox instead.

File: src/components/pages/analytics-page/analytics-page.styles.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    },
+  },
+  workspaceName: {
+    fontSize: theme.typography.h6.fontSize,
+    fontWeight: 'bold',
+    marginBottom: '2px',
+    // whiteSpace: 'nowrap',
+    // overflow: 'hidden',
+    // textOverflow: 'ellipsis',
+    // maxWidth: '100%',
```
Comment (AnthonyWhitaker) [inline src/components/pages/analytics-page/analytics-page.styles.ts:66]: Remove comments?

File: src/components/pages/role-management-page/edit-role-dialog.tsx [hunk 2]
```diff
@@ -77,25 +86,32 @@ export const EditRoleDialog = (props: EditRoleDialogProps) => {
...
+      const workspace = workspaces.find(x => x.id === workspaceId);
+      if (!workspace) {
+        throw new Error('Unable to find workspace!');
+      }
+
+      return workspace;
+    });
+
+    const containsPii = newSelectedWorkspaces.filter(x => x.pii).length > 0;
+    const containsPhi = newSelectedWorkspaces.filter(x => x.phi).length > 0;
```
Comment (AnthonyWhitaker) [inline src/components/pages/role-management-page/edit-role-dialog.tsx:101]:
```
    const containsPii = newSelectedWorkspaces.some(x => x.pii);
    const containsPhi = newSelectedWorkspaces.some(x => x.phi);
```

File: src/components/typography-truncated/typography-truncated.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  lines: number,
+  lineHeight?: number,
+};
+
+export const TypographyTruncated = (props: TypographyTruncatedProps) => {
+  const { children, lines, lineHeight, ...rest } = props;
+  const classes = useStyles({
+    lines,
+    lineHeight,
+  });
```
Comment (AnthonyWhitaker) [inline src/components/typography-truncated/typography-truncated.tsx:19]:
```suggestion
    variant: props.variant,
  });
```

File: src/components/typography-truncated/typography-truncated.styles.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+export default (args: {
+  lines: number,
+  lineHeight?: number,
+}) => {
+  const styles = makeStyles((theme: Theme) => {
+    const {
+      lines,
+      lineHeight = theme.typography.body1.lineHeight,
+    } = args;
```
Comment (AnthonyWhitaker) [inline src/components/typography-truncated/typography-truncated.styles.ts:13]:
```suggestion
import { makeStyles } from '@material-ui/styles';

export default (args: {
  lines: number,
  lineHeight?: number,
}) => {
  const styles = makeStyles((theme: Theme) => {
    const {
      lines,
      lineHeight = theme.typography.body1.lineHeight,
    } = args;
```

File: src/components/typography-truncated/typography-truncated.styles.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { createStyles } from '@material-ui/core';
+import { Theme } from '@material-ui/core/styles';
+import { makeStyles } from '@material-ui/styles';
+
```
Comment (AnthonyWhitaker) [inline src/components/typography-truncated/typography-truncated.styles.ts:4]:
```suggestion
import { Variant } from '@material-ui/core/styles/createTypography';

````

File: src/components/typography-truncated/typography-truncated.styles.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+export default (args: {
+  lines: number,
+  lineHeight?: number,
+}) => {
```
Comment (AnthonyWhitaker) [inline src/components/typography-truncated/typography-truncated.styles.ts:8]:
```suggestion
    variant?: Variant,
}) => {
```

File: src/components/typography-truncated/typography-truncated.styles.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  const styles = makeStyles((theme: Theme) => {
+    const {
+      lines,
+      lineHeight = theme.typography.body1.lineHeight,
```
Comment (AnthonyWhitaker) [inline src/components/typography-truncated/typography-truncated.styles.ts:12]:
```suggestion
      lineHeight = theme.typography[args.variant ?? 'body1'].lineHeight,
```

**Review by AnthonyWhitaker — APPROVED** [review]
This is good stuff!

**Pull request merged**